netty/src/main/java/io/grpc/netty/NettyServerBuilder.java

```diff
@@ -180,6 +181,18 @@ public NettyServerBuilder sslContext(SslContext sslContext) {
   }
 
   /**
+   * Sets the {@link ProtocolNegotiator} to be used. If non-{@code null}, overrides the value
+   * specified in {@link #sslContext(SslContext)}.
+   *
+   * <p>Default: {@code null}.
+   */
+  public final NettyServerBuilder protocolNegotiator(
```

Comment:
What? Why is it "essentially part of the builder API now and should assume the same strictness of usage." That argument seems circular since it is being added right now...

It is _definitely_ less intended to be used than other fields. The sslContext, event loops, max message size, etc are all quite reasonable for someone to use. They are intended for people to use. This is intended for a transport author to use; mere mortals _should not_ use it, because it is essentially guaranteed that it won't work properly (at least on client-side), given how much trouble _we've_ had implementing them.